src/Tasks/Microsoft.NET.Build.Tasks/CompileReadyToRunImages.cs

```diff
@@ -0,0 +1,415 @@
+                        file.ItemSpec = outputR2RImage;
+
+                        // Note: ReadyToRun PDB/Map files are not needed for debugging. They are only used for profiling, therefore the default behavior is to not generate them
+                        // unless an explicit ReadyToRunEmitSymbols flag is enabled by the app developer. There is also another way to profile that the runtime supports, which does
+                        // not rely on the native PDBs/Map files, so creating them is really an opt-in option, typically used by advanced users.
+                        // For debugging, only the IL PDBs are required.
+                        if (ReadyToRunEmitSymbols)
+                        {
+                            string outputPDBImage;
+                            if (CreatePDBForImage(file, out outputPDBImage))
+                            {
+                                // If we create a native PDB/Map image, we add it to the publishing list in addition to all other files. Native PDBs/Maps do not replace IL PDBs.
+                                TaskItem newPDBfile = new TaskItem();
+                                newPDBfile.ItemSpec = outputPDBImage;
+                                newPDBfile.SetMetadata(MetadataKeys.CopyToPublishDirectory, file.GetMetadata(MetadataKeys.CopyToPublishDirectory));
+                                newPDBfile.SetMetadata(MetadataKeys.RelativePath, Path.GetFileName(outputPDBImage));
+                                newPDBfile.RemoveMetadata(MetadataKeys.OriginalItemSpec);
+
+                                outputFiles.Add(newPDBfile);
+                            }
+                        }
+                    }
+                }
+
+                outputFiles.Add(file);
+            }
+        }
+
+        bool InputFileEligibleForCompilation(ITaskItem file)
+        {
+            // Check if the file is explicitly excluded from being compiled
+            if (ReadyToRunExcludeList != null)
+            {
+                foreach (var item in ReadyToRunExcludeList)
+                    if (String.Compare(Path.GetFileName(file.ItemSpec), item, true) == 0)
+                        return false;
+            }
+
+            // Check to see if this is a valid ILOnly image that we can compile
+            using (FileStream fs = new FileStream(file.ItemSpec, FileMode.Open, FileAccess.Read))
+            {
+                PEReader pereader;
+                MetadataReader mdReader;
+
+                try
+                {
+                    pereader = new PEReader(fs);
+
+                    if (pereader.PEHeaders == null || pereader.PEHeaders.CorHeader == null)
+                        return false;
+
+                    if ((pereader.PEHeaders.CorHeader.Flags & CorFlags.ILOnly) != CorFlags.ILOnly)
+                        return false;
+
+                    mdReader = pereader.GetMetadataReader();
+                }
+                catch
```

Comment:
Use `catch (BadImageFormatException)` here.

Reply:
is that the only kind of exception we can get if the file is not a valid PE file?

Reply:
Yes. I worked on System.Reflection.Metadata. We have the same pattern elsewhere. I'd rather risk a bug that is easy to fix (add an exception type here) then hide bugs with catch all.

Reply:
But, really, if you get something other than BadImageFormatException for invalid PE file, it's either a bug in S.R.M or something catastrophic. Neither of those should be masked here.

Reply:
Verified that there are no other exception types thrown on non-PE files.